src/XRP/xrp-transaction.ts

```diff
@@ -0,0 +1,109 @@
+import { Transaction } from '../generated/web/org/xrpl/rpc/v1/transaction_pb'
+import RippledFlags from '../rippled-flags'
+import XRPSigner from './xrp-signer'
+import XRPTransactionType from './xrp-transaction-type'
+import XRPPayment from './xrp-payment'
+import XRPMemo from './xrp-memo'
+
+/* A transaction on the XRP Ledger.
+ *
+ * @see: https://xrpl.org/transaction-formats.html
+ */
+// TODO(amiecorso): Modify this object to use X-Address format.
+export default class XRPTransaction {
+  public static from(transaction: Transaction): XRPTransaction | undefined {
+    const account = transaction
+      .getAccount()
+      ?.getValue()
+      ?.getAddress()
+    const fee = transaction.getFee()?.getDrops()
+    const sequence = transaction.getSequence()?.getValue()
+    const signingPublicKey = transaction.getSigningPublicKey()?.getValue()
+    const transactionSignature = transaction
+      .getTransactionSignature()
+      ?.getValue()
+
+    const accountTransactionID = transaction
+      .getAccountTransactionId()
+      ?.getValue()
+    const rawValue = transaction.getFlags()?.getValue()
+    let flags
+    if (rawValue) {
+      flags = RippledFlags.checkFlag(rawValue, RippledFlags.TF_PARTIAL_PAYMENT)
```

Comment:
This method will check if the partial payment flag is set in the `rawValue`. 
```
 RippledFlags.checkFlag(rawValue, RippledFlags.TF_PARTIAL_PAYMENT)
```

You really want to pass along this value, which you could just do as:
```
const flags = transaction.getFlags()?.getValue()
```

In swift, we make flags a struct because Swift has some built in magic for working with bit sets. In JS it seems the best way is to just work with raw numbers.

Reply:
hmm, I don't totally understand this second part. are you saying that ```const flags = transaction.getFlags()?.getValue()``` is sufficient and I don't need the ```if (rawValue) { 
flags = RippledFlags.checkFlag(....)``` ?

Reply:
correct